test/src/telemetry/test_telemetry.c

```diff
@@ -193,68 +193,15 @@ test_telemetry_parse_version(PG_FUNCTION_ARGS)
 	return HeapTupleGetDatum(tuple);
 }
 
+/* Try to get the telemetry function to handle errors. Never connect to the actual endpoint. Only test cases that will result in connection errors. */
 Datum
-test_telemetry(PG_FUNCTION_ARGS)
+test_telemetry_main_conn(PG_FUNCTION_ARGS)
 {
-	Connection *conn;
```

Comment:
I didn't remove it! It just restructured it to call `telemetry_main` directly (and in doing so, I needed to change the signature of `telemetry_main`). Otherwise, it's ugly to copy-paste all the logic from `telemetry_main` into this function; in fact, originally, the code was copy-pasted incorrectly, which is another reason why we didn't catch this bug earlier.

In fact, this function is now called in the regression tests, because I specifically pass parameters that will result in a NULL connection, which means we won't poke our endpoint.

Reply:
But it is not returning anything useful any more, so it doesn't do the same thing as you cannot inspect the response. And, you cannot specify the port, so you need to be super user to run `netcat` on port 80.

Also, why change the name of the function if this was just a refactor?

You can just add a new test function for calling `telemetry_main()`, or properly refactor `telemetry_main()` to take also the port and return the response.

Reply:
I'll refactor since inspecting the response is nice. But for the port, can't you just specify port by specifying `service`, or is that a hack? In particular, in my local `/etc/services` file, looks like `http-alt` is port 8080.